servicetalk-http-api/src/main/java/io/servicetalk/http/api/ExecutionContextAwareHttpBuilder.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import io.servicetalk.buffer.api.BufferAllocator;
+import io.servicetalk.transport.api.IoExecutor;
+
+public interface ExecutionContextAwareHttpBuilder<ResolvedAddress> {
```

Comment:
Can this interface be package private. Ideally we can minimize complexity related to class hierarchy of builders. If not, can you add javadoc?

Reply:
That's an omission, should definitely be package private. I'll skip the javadoc as it's not intended to be exposed to users.

Reply:
Re: minimize complexity – I believe short, domain oriented interfaces achieve that purpose. I didn't minimise the amount of interfaces, but rather introduced more decoupling and specialization of interfaces.
As a result of my changes, abstract class hierarchy:
    BaseHttpBuilder <- HttpClientBuilder <- BaseSingleAddressHttpClientBuilder
can be squashed to one abstract class. I'm not sure whether that's what we in fact need, but merging `BaseHttpBuilder` with `HttpClientBuilder` sounds reasonable. WDYT?

Reply:
+1 if we can reduce the hierarchy depth and additional types